concent_api/core/message_handlers.py

```diff
@@ -272,6 +275,7 @@ def handle_send_reject_report_computed_task(client_message):
 
     # RejectReportComputedTask should contain empty cannot_compute_task and task_failure
     else:
+        assert client_message.reason == message.RejectReportComputedTask.REASON.SubtaskTimeLimitExceeded
```

Comment:
I'm not sure about that. What's the point of adding validation above just to have assert below that limit our code to those three reasons that `RejectReportComputedTask` currently has. If golem add some new reasons in future, our code will crush on that assert.
@rwrzesien , @cameel ?

Reply:
> If golem add some new reasons in future, our code will crush on that assert.

That's exactly why it's here. If Golem adds a new reason and we don't update our code to handle it, we have a bug. The current code only works for the three reasons we currently know. This assert can detect such a bug.

The problem is that someone can silently break our code by doing a change in some far away place in another library. We don't want that. It's better to crash than to ignore such an error. Also, this crash is meant to happen in development. When you update golem-messages you're supposed to see it and fix the problem.

We need to be doing it more. Basically any final `else:` in an `if` that's defined for all values of an enum should have such an assert.